pkg/cloud/rgraph/rnode/backendservice/node.go

```diff
@@ -38,6 +38,8 @@ var _ rnode.Node = (*backendServiceNode)(nil)
 
 func (n *backendServiceNode) Resource() rnode.UntypedResource { return n.resource }
 
+func (n *backendServiceNode) BackendResource() BackendService { return n.resource }
```

Comment:
Two things:

1. Why not cast Resource() to the concrete type in the test? Why do we have to wrap up the cast with a helper method? If this is just for testing, I think it doesn't make sense to do this...

2. If this is a pattern we want to follow, I recommend naming this `TypedResource()` and implementing it on every resource, not just the BackendService.

We should always make the code as uniform as possible -- unfortunately Golang does not allow us to implement these kinds of uniformity constraints in the type system itself, but we should always be looking for ways to make things the same across _all_ of the types to make it easier for the user of the library to work with. 

This includes patterns such as this.

Reply:
sorry this is a leftover form other tests it should not be in this review